capdl-loader-app/helpers.cmake

```diff
@@ -28,10 +28,7 @@ function(BuildCapDLApplication)
     )
 
     if(DEFINED platform_yaml)
-        set(
-            PLATFORM_SIFT
-            "${CMAKE_SOURCE_DIR}/../../tools/seL4/cmake-tool/helpers/platform_sift.py"
-        )
+        set(PLATFORM_SIFT "${SEL4_CMAKE_TOOL_DIR}/helpers/platform_sift.py")
```

Comment:
@axel-h where is SEL4_CMAKE_TOOL_DIR defined?

Reply:
Good catch, this is set up in a custom CMake file actually. I've updated the patch to define this internally if it is not defined. Do you have a better suggestion where to define this?

Reply:
Normally this would be defined in the Find{libname}.cmake file, but this wasn't created for the sel4_tools/cmake-tool repository.  Instead cmake-tool/helpers/ is directly added to the CMAKE_MODULE_PATH path with the expectation that any script in this repo will have an accompanying .cmake file that will define the relevant variables. I think because platform_sift.py was added for the Elfloader originally it got through by having a relative include path because they are both in the same repository.
For now, this should work and has the same effect as if platform_sift.py was a cmake file:
`find_file(PLATFORM_SIFT platform_sift.py PATHS ${CMAKE_MODULE_PATH} NO_CMAKE_FIND_ROOT_PATH)`

Reply:
(It's also good practice to add the following after too:
```
mark_as_advanced(FORCE PLATFORM_SIFT)
if("${PLATFORM_SIFT}" STREQUAL "PLATFORM_SIFT-NOTFOUND")
    message(FATAL_ERROR "Failed to find platform_sift.py. Consider cmake -DPLATFORM_SIFT=/path/to/file")
endif()
```
)

Reply:
thanks, I've added this also now.